service/api_logic/sports_logic.py

```diff
@@ -1,10 +1,9 @@
-from database.models import Sport
-from database.session import SessionLocal
+from database.models import Sport, League
 from exept.colors_text import print_error_message
+from exept.handle_exeptions import handle_exceptions
 
-session = SessionLocal()
-
-def get_all_sports():
+@handle_exceptions
+def get_all_sports(session):
     sports = session.query(Sport).all()
     if not sports:
         print_error_message("No sports found in the database")
```

Comment:
It'd be better to raise a NotFound Exception here (you already have one. Some tuning and it will work here too)

Also, service layer (Business logic) does not care about API. It must handle its own tasks and that's it. So do not return status code here - it should be handled on API level (plus: if you raise a custom exception here there won't be any need for that return statement whatsoever)

And I'm not sure about 204 status code. It probably must've been 404.

Summary:
Throw a custom exception